tests/components/swiss_public_transport/test_config_flow.py

```diff
@@ -0,0 +1,228 @@
+"""Test the swiss_public_transport config flow."""
+from unittest.mock import AsyncMock
+
+import aiohttp
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.swiss_public_transport import config_flow
+from homeassistant.components.swiss_public_transport.const import (
+    CONF_DESTINATION,
+    CONF_START,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_flow_user_init(hass: HomeAssistant, aioclient_mock: AiohttpClientMocker):
+    """Test the initialization of the form in the first step of the config flow."""
+    aioclient_mock.get(
+        "http://transport.opendata.ch/v1/connections?from=test_start&to=test_destination",
+        json={"from": "test_start", "to": "test_destination", "connections": []},
+    )
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
```

Comment:
Can we instead of mocking the http call mock the library? This way we're only testing the HA code